crates/astria-composer/src/executor/mod.rs

```diff
@@ -607,6 +557,72 @@ async fn submit_tx(
     res
 }
 
+/// Reports and logs exit reason
+#[instrument(skip_all, err)]
+fn report_exit(reason: &eyre::Result<&str>) -> eyre::Result<()> {
```

Comment:
Created new `utils` module with this functionality. I also renamed the function to `log_exit`, since I thought this was more accurate and less likely to be confused with `composer::report_exit` which functions slightly differently.